src/Provider/ReflectionProvider.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Provider;
+
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\ReflectionProvider as PHPStanReflectionProvider;
+
+final class ReflectionProvider
+{
+    public function __construct(
+        private readonly PHPStanReflectionProvider $reflectionProvider,
+        private readonly CurrentFileProvider $currentFileProvider
+    ) {
+    }
+
+    public function __call($name, $params)
+    {
+        return $this->reflectionProvider->$name(...$params);
+    }
+
+    public function getClass(string $className): ClassReflection
+    {
+        if (! $this->reflectionProvider->hasClass($className)) {
+            dump('here');
+        }
+
+        return $this->reflectionProvider->getClass($className);
+    }
```

Comment:
@TomasVotruba I am trying to override PHPStan's `ReflectionProvider::getClass()` method, any way to create `ClassReflection` instance manually based on the file/nodes?

Reply:
I never tried to fake my own `ClassReflection` of PHPStan, so I'm not sure how to do that.

Reply:
Fixed 🎉 by fallback to non-parallel when got error Class not found https://github.com/rectorphp/rector-src/pull/1613/commits/454acbcbe93b647a4996776267721da2e6407b5c